### PR TITLE
sink(ticdc): ignore sync.runtime_Semacquire in mysql leak test

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: ref #5356

### What is changed and how it works?
1. Temporarily ignore sync.runtime_Semacquire in mysql leak test.
2. Enable t.Parallel for cdc/sink/mysql, note that some tests which modify global variables cannot be run in parallel.

### Check List 

Tests 

 - Unit test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by zhaoxinyu — APPROVED** [review]

File: cdc/sink/mysql/causality_test.go [hunk 1]
```diff
@@ -69,6 +71,7 @@ func TestCausality(t *testing.T) {
 }
 
 func TestGenKeys(t *testing.T) {
+	defer leakutil.VerifyNone(t)
```
Comment (maxshuang) [inline cdc/sink/mysql/causality_test.go:74]: can u explain why we can use t.Parallel() and VerifyTestMain() in mysql related package?

Comment (CharlesCheung96) [reply cdc/sink/mysql/causality_test.go:74]: VerifyNone is used to identify which tests have coroutine leaks, ref: https://github.com/pingcap/tiflow/issues/5356#issuecomment-1123213475. The root cause is that mysql Sink was not reliably closed. After the PR was fixed, we were able to open `t.Parallel`

**Review by asddongmen — APPROVED** [review]

**Review by maxshuang — APPROVED** [review]

**Pull request merged**